packages/gatsby-theme-aio/algolia/records/record-utils.js

```diff
@@ -0,0 +1,55 @@
+const AlgoliaHTMLExtractor = require('algolia-html-extractor');
+const htmlExtractor = new AlgoliaHTMLExtractor();
+const { selectAll } = require('unist-util-select');
+const { v4: uuidv4 } = require('uuid');
+
+const createRawRecords = (node, options, fileContent = null) => {
+  if (fileContent != null) {
+    return htmlExtractor
+      .run(fileContent, { cssSelector: options.tagsToIndex })
+      .filter(record => record.content.length >= options.minCharsLengthPerTag);
+  } else {
+    // https://mdxjs.com/table-of-components
+    return selectAll(options.tagsToIndex, node.mdxAST).filter(
+      record => record.value.length >= options.minCharsLengthPerTag
+    );
+  }
+};
+
+const createAlgoliaRecords = (node, records) => {
+  let { mdxAST, objectID, title, slug, headings, wordCount, ...restNodeFields } = node;
+
+  return records.map(record => ({
+    objectID: record.objectID ?? uuidv4(record.value.toString()),
+    title: getTitle(node),
+    ...restNodeFields,
+    // headings: headings.map((heading) => heading.value),
```

Comment:
why is this commented out ?

Reply:
Ah, yeah. We don't need headings from the node anymore because previousHeadings and contentHeading give us more precisely what we need for the parsed paragraphs, code, tableCells, and other parsed content. So I left it there to remind myself to think more about the `getHeadings()` implementation. I'll replace it with a TODO instead.